components/finance/FinanceCategories/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { gql, useQuery } from '@apollo/client'
+import React, { useState } from 'react'
+
+import Loader from '../../common/Loader'
+
+const FinanceCategories = () => {
+  const getCategories = () => gql`
+    query User {
+      user {
+        userSettings {
+          workItems {
+            name
+            incomeOrExpense
+            totalAmount {
+              type
+              amount
+            }
+          }
+        }
+      }
+    }
+  `
+  const [activeItemIndex, setActiveItemIndex] = useState(0)
+  const { loading, error, data } = useQuery(getCategories())
+
+  if (loading || error)
+    return (
+      <div>
+        <Loader open={loading} error={error} />
+      </div>
+    )
+
+  const { workItems } = data.user.userSettings
+  return (
+    <>
+      <div className="ml-1">Income/Expense</div>
+      <ul className="flex flex-col w-96 mt-0">
+        {workItems.map((item: any, i: number) => (
+          // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
+          <li key={item.name} className="flex mt-px" onMouseOver={() => setActiveItemIndex(i)}>
```

Comment:
create each line item as a separate component so that the logic can be reused when we have other lists like bank accounts, credit cards etc

Reply:
made the required changes